synapse/app/generic_worker.py

```diff
@@ -459,6 +459,7 @@ def start(config_options: List[str]) -> None:
 
     synapse.events.USE_FROZEN_DICTS = config.server.use_frozen_dicts
     synapse.util.caches.TRACK_MEMORY_USAGE = config.caches.track_memory_usage
+    config.caches.resize_all_caches()
```

Comment:
Doesn't this need to be in the `synapse/app/_base.py` so that the main process picks it up too?

Reply:
oh ffs. It was meant to be. thanks

Reply:
See 55963acf1. I also snuck in f199d82be.